bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/handler/MPDHandler.java

```diff
@@ -0,0 +1,292 @@
+    }
+
+    private void handlePlayerCommand(String channelId, Command command) {
+        switch (channelId) {
+            case CHANNEL_CONTROL:
+                handleCommandControl(command);
+                break;
+            case CHANNEL_STOP:
+                handleCommandStop(command);
+                break;
+            case CHANNEL_VOLUME:
+                handleCommandVolume(command);
+                break;
+        }
+    }
+
+    private void handleCommandControl(Command command) {
+        if (command instanceof PlayPauseType) {
+            if (command == PlayPauseType.PLAY) {
+                connection.play();
+            } else if (command == PlayPauseType.PAUSE) {
+                connection.pause();
+            }
+        } else if (command instanceof NextPreviousType) {
+            if (command == NextPreviousType.NEXT) {
+                connection.playNext();
+            } else if (command == NextPreviousType.PREVIOUS) {
+                connection.playPrevious();
+            }
+        } else {
+            // Rewind and Fast Forward are currently not implemented by the binding
+            logger.debug("Control command {} is not supported", command);
+        }
+    }
+
+    private void handleCommandStop(Command command) {
+        if (command instanceof OnOffType) {
+            if (command == OnOffType.ON) {
+                connection.stop();
+            } else if (command == OnOffType.OFF) {
+                connection.play();
+            }
+        } else {
+            logger.debug("Stop Command {} is not supported", command);
+            return;
+        }
+    }
+
+    private void handleCommandVolume(Command command) {
+        int newValue = 0;
+        if (command instanceof IncreaseDecreaseType) {
+            if (command == IncreaseDecreaseType.INCREASE) {
+                newValue = Math.min(100, volume + 1);
+            } else if (command == IncreaseDecreaseType.DECREASE) {
+                newValue = Math.max(0, volume - 1);
+            }
+        } else if (command instanceof OnOffType) {
+            if (command == OnOffType.ON) {
+                newValue = 100;
+            } else if (command == OnOffType.OFF) {
+                newValue = 0;
+            }
+        } else if (command instanceof DecimalType) {
+            newValue = ((DecimalType) command).intValue();
+        } else if (command instanceof PercentType) {
+            newValue = ((PercentType) command).intValue();
+        } else {
+            logger.debug("Command {} is not supported to change volume", command);
+            return;
+        }
+
+        connection.setVolume(newValue);
+    }
+
+    private void updateChannel(String channelID, State state) {
+        State previousState = stateMap.put(channelID, state);
+        if (previousState == null || !previousState.equals(state)) {
+            updateState(channelID, state);
+        }
+    }
+
+    @Override
+    public void updateStatus(MPDStatus status) {
```

Comment:
Remark: This method-name could be slightly confusing because this class also has `updateStatus` which updates the thing status.

Reply:
I have renamed it to updateMPDStatus, as well as updateSong to updateMPDSong.